ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts

```diff
@@ -59,7 +60,9 @@ export class SessionWillExpireComponent implements OnInit, OnDestroy {
                 this.timeLeft = `${formattedMinutes}:${formattedSeconds}`;
             } else {
                 this.timeLeft = '00:00';
-                this.auth0.logout('session-expired');
+                if (!this.blockUI) {
+                    this.auth0.logout('session-expired');
+                }
```

Comment:
If a user clicked 'Continue' or 'Sign Out' buttons Auth0 needs some time to handle the user's request to refresh the session or redirect the user to the home page. While Auth0 is handling the request, time may end even and will be sent one more request. To prevent it, let's check whether the user chose something or not.